script/20240131-maptoken-pixel/20240131-maptoken-pixel-mainchain.s.sol

```diff
@@ -0,0 +1,142 @@
+import { Contract } from "../utils/Contract.sol";
+import { BridgeMigration } from "../BridgeMigration.sol";
+import { Network } from "../utils/Network.sol";
+import { Contract } from "../utils/Contract.sol";
+import { IGeneralConfigExtended } from "../IGeneralConfigExtended.sol";
+
+import "./maptoken-pixel-configs.s.sol";
+import "./update-axiechat-config.s.sol";
+
+contract Migration__20240131_MapTokenPixelMainchain is BridgeMigration, Migration__MapToken_Pixel_Config, Migration__Update_AxieChat_Config {
+  RoninBridgeManager internal _roninBridgeManager;
+  address internal _mainchainGatewayV3;
+  address internal _mainchainBridgeManager;
+
+  function setUp() public override {
+    super.setUp();
+
+    _roninBridgeManager = RoninBridgeManager(_config.getAddressFromCurrentNetwork(Contract.RoninBridgeManager.key()));
+    _mainchainGatewayV3 = _config.getAddress(
+      _config.getCompanionNetwork(_config.getNetworkByChainId(block.chainid)).key(),
+      Contract.MainchainGatewayV3.key()
+    );
+    _mainchainBridgeManager = _config.getAddress(
+      _config.getCompanionNetwork(_config.getNetworkByChainId(block.chainid)).key(),
+      Contract.MainchainBridgeManager.key()
+    );
+  }
+
+  function run() public {
+    address[] memory mainchainTokens = new address[](1);
+    address[] memory roninTokens = new address[](1);
+    Token.Standard[] memory standards = new Token.Standard[](1);
+    uint256[][4] memory thresholds;
+
+    uint256 expiredTime = block.timestamp + 10 days;
+    address[] memory targets = new address[](4);
+    uint256[] memory values = new uint256[](4);
+    bytes[] memory calldatas = new bytes[](4);
+    uint256[] memory gasAmounts = new uint256[](4);
+
+    // ================ PIXEL ERC-20 ======================
+
+    mainchainTokens[0] = _pixelMainchainToken;
+    roninTokens[0] = _pixelRoninToken;
+    standards[0] = Token.Standard.ERC20;
+    // highTierThreshold
+    thresholds[0] = new uint256[](1);
+    thresholds[0][0] = _highTierThreshold;
+    // lockedThreshold
+    thresholds[1] = new uint256[](1);
+    thresholds[1][0] = _lockedThreshold;
+    // unlockFeePercentages
+    thresholds[2] = new uint256[](1);
+    thresholds[2][0] = _unlockFeePercentages;
+    // dailyWithdrawalLimit
+    thresholds[3] = new uint256[](1);
+    thresholds[3][0] = _dailyWithdrawalLimit;
+
+    // function mapTokensAndThresholds(
+    //   address[] calldata _mainchainTokens,
+    //   address[] calldata _roninTokens,
+    //   Token.Standard[] calldata _standards,
+    //   uint256[][4] calldata _thresholds
+    // )
+
+    bytes memory innerData = abi.encodeCall(IMainchainGatewayV3.mapTokensAndThresholds, (
+      mainchainTokens,
+      roninTokens,
+      standards,
+      thresholds
+    ));
+
+    bytes memory proxyData = abi.encodeWithSignature("functionDelegateCall(bytes)", innerData);
+
+    targets[0] = _mainchainGatewayV3;
+    values[0] = 0;
+    calldatas[0] = proxyData;
+    gasAmounts[0] = 1_000_000;
```

Comment:
1. Not sure 1e6 is enough
2. Please consider improving this model while revamp the bridge manager

Reply:
1. Enough, already checked in simulation
2. I don't get it

Reply:
AGG also use 1M gas